### PR TITLE
[Core] turn FilesystemExtensions into class

### PR DESCRIPTION
**Pull request by philbucher**
since we prefer to use classes with static methods over namespaces I thought I could clean this up

### CODE REVIEW

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 // System includes
 #include <fstream>
+#include <tuple>
```
Comment (matekelemen) [inline kratos/tests/cpp_tests/sources/test_filesystem.cpp:16]: tuple?

**Review by matekelemen — COMMENTED** [review]
lgtm ®

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/tests/cpp_tests/sources/test_filesystem.cpp:16]: for std::ignore

**Review by rubenzorrilla — APPROVED** [review]
Thanks!

**Pull request merged**